core/src/internal_flags.rs

```diff
@@ -29,6 +29,15 @@ pub(crate) enum CoreInternalFlags {
     UpsertSearchAttributeOnPatch = 2,
     /// We received a value higher than this code can understand.
     TooHigh = u32::MAX,
+    /// Prior to this flag, we truncated commands received from lang at the
+    /// first terminal (i.e. workflow-terminating) command. With this flag, we
+    /// reorder commands such that all non-terminal commands come first,
+    /// followed by the first terminal command, if any (it's possible that
+    /// multiple workflow coroutines generated a terminal command). This has the
+    /// consequence that all non-terminal commands are sent to the server, even
+    /// if in the sequence delivered by lang they came after a terminal command.
+    /// See https://github.com/temporalio/features/issues/481.
+    ReorderCommands = 3,
```

Comment:
Can we move this above `TooHigh`? I might also call it `ReorderTerminalCommands` just to be a bit more explicit